libdevcore/FileSystem.cpp

```diff
@@ -55,15 +55,11 @@ void dev::setIpcPath(string const& _ipcDir)
 
 string dev::getIpcPath()
 {
-	if (s_ethereumIpcPath.empty())
-		return string(getDataDir());
-	else
-	{
-		size_t socketPos = s_ethereumIpcPath.rfind("geth.ipc");
-		if (socketPos != string::npos)
-			return s_ethereumIpcPath.substr(0, socketPos);
-		return s_ethereumIpcPath;
-	}
+	// Strip "geth.ipc" suffix if provided.
+	size_t socketPos = s_ethereumIpcPath.rfind("geth.ipc");
+	if (socketPos != string::npos)
```

Comment:
`substr(0, string::npos)` will return the whole string, so I think this if is not needed